tests/manager/celery/tasks/test_axis.py

```diff
@@ -0,0 +1,401 @@
+
+    ts = timestamp(
+        _db=db.session,
+        collection=collection,
+        service_name="palace.manager.celery.tasks.axis.list_identifiers_for_import",
+        default_start_time=DEFAULT_START_TIME,
+    )
+
+    assert ts.start and ts.start > current_time
+    assert not queue_collection_import_lock_fixture.task_lock.locked()
+    assert mock_api.recent_activity.call_count == 1
+    assert mock_api.recent_activity.call_args[0][0] == axis.DEFAULT_START_TIME
+    assert "Finished listing identifiers in collection" in caplog.text
+
+
+def generate_test_metadata_and_circulation_objects(test_ids):
+    metadata_and_circulation_data_list = []
+    for id in test_ids:
+        data_source = "data_source"
+        identifier = IdentifierData(type=Identifier.AXIS_360_ID, identifier=id)
+        metadata_and_circulation_data_list.append(
+            (
+                Metadata(data_source=data_source, primary_identifier=identifier),
+                CirculationData(data_source=data_source, primary_identifier=identifier),
+            )
+        )
+    return metadata_and_circulation_data_list
+
+
+def test_import_items(
+    db: DatabaseTransactionFixture,
+    celery_fixture: CeleryFixture,
+    queue_collection_import_lock_fixture: QueueCollectionImportLockFixture,
+    caplog: pytest.LogCaptureFixture,
+):
+    set_caplog_level_to_info(caplog)
+    collection = db.collection(name="test_collection", protocol=Axis360API.label())
+
+    mock_api = MagicMock()
+    with (
+        patch.object(axis, "create_api") as mock_create_api,
+        patch.object(axis, "requeue_import_identifiers_task") as requeue,
+    ):
+        mock_create_api.return_value = mock_api
+        edition_1, lp_1 = db.edition(with_license_pool=True)
+        edition_2, lp_2 = db.edition(with_license_pool=True)
+        title_ids = [x.primary_identifier.identifier for x in [edition_1, edition_2]]
+        mock_api.availability_by_title_ids.return_value = (
+            generate_test_metadata_and_circulation_objects(title_ids)
+        )
+        identifiers = [x.primary_identifier.identifier for x in [edition_1, edition_2]]
+        mock_api.update_book.side_effect = [
+            (edition_1, False, lp_1, False),
+            (edition_2, False, lp_2, False),
+        ]
+        import_identifiers.delay(
+            collection.id, identifiers=identifiers, batch_size=25
+        ).wait()
+
+    assert mock_api.availability_by_title_ids.call_count == 1
+    assert mock_api.availability_by_title_ids.call_args.kwargs["title_ids"] == title_ids
+    assert mock_api.update_book.call_count == 2
+    assert requeue.call_count == 0
+
+    assert f"Edition (id={edition_1.id}" in caplog.text
+    assert (
+        f"Finished run importing identifiers for collection ({collection.name}, id={collection.id})"
+        in caplog.text
+    )
+    assert f"Imported {len(title_ids)} identifiers" in caplog.text
+
+
+def test_import_items_with_requeue(
```

Comment:
I'd rather see this test use less mocks, and actually allow the task to be re-queued, so we can test the full behavior when the task has to cycle more then once. This will be easier once you switch to using `task.replace`.